### PR TITLE
Add CollectionsRail tracking and tests

### PR DESCRIPTION
**Pull request by eessex**
HOLD for confirmation of tracking calls from @abhitip 

- Adds tracking events for Collections Rail impressions and clicks
- Adds tests for Collections Rail components and tracking events
- Renames `CollectionEntity` file to match component name
- Removes extra responsive declarations

### CODE REVIEW

**Comment by eessex** [comment]
@xtina-starr ok to merge if it is easier to do additional work for queries with this included.  Put the HOLD notice because there is some ongoing discussion of whether we should use `CollectionsRail`, `collectionsRail` or `Collections Rail` for the tracking call.  Can update in a separate PR as well.

File: src/Components/CollectionsRail/CollectionsRail.tsx [hunk 1]
```diff
@@ -21,38 +25,52 @@ const RailsWrapper = styled(Flex)`
...
-    padding: 40px;
-  `};
-  ${pMedia.lg`
-    padding: 40px;
+    padding: ${space(4)}px;
   `};
-  ${pMedia.xs`
-    padding: 40px 30px;
+  ${pMedia.sm`
+    padding: ${space(4)}px ${space(2)}px;
```
Comment (eessex) [inline src/Components/CollectionsRail/CollectionsRail.tsx]: Made some minor updates here to remove extra declaration, and match padding to `RelatedArticlesCanvas`

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit f740c135ab2a88d93b55480bc2d0826fd96fd8f8

https://deploy-preview-2068--artsy-reaction.netlify.com

**Comment by xtina-starr** [comment]
I think we can go ahead and merge this. If there is a change I can probably add that to the PR that updates the query in reaction or it can be a separate PR all together. 

Do you want to update the title and go ahead and merge?

**Comment by artsyit** [comment]
:rocket: PR was released in v14.1.13 :rocket:

**Pull request merged**